test-framework/arquillian/src/main/java/io/quarkus/arquillian/MethodParameterInjectionPoint.java

```diff
@@ -0,0 +1,118 @@
+package io.quarkus.arquillian;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.Annotated;
+import javax.enterprise.inject.spi.AnnotatedCallable;
+import javax.enterprise.inject.spi.AnnotatedParameter;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Qualifier;
+
+public class MethodParameterInjectionPoint<T> implements InjectionPoint {
+    private Method method;
+    private int position;
+    private BeanManager beanManager;
+
+    public MethodParameterInjectionPoint(Method method, int position, BeanManager beanManager) {
+        this.method = method;
+        this.position = position;
+        this.beanManager = beanManager;
+    }
+
+    public Bean<?> getBean() {
+        return null;
+    }
+
+    public Member getMember() {
+        return method;
+    }
+
+    public Set<Annotation> getQualifiers() {
+        Set<Annotation> qualifiers = new HashSet<>();
+        for (Annotation potentialQualifier : method.getParameterAnnotations()[position]) {
+            if (potentialQualifier.annotationType().getAnnotation(Qualifier.class) != null) {
+                qualifiers.add(potentialQualifier);
+            }
+        }
+        if (qualifiers.size() == 0) {
+            qualifiers.add(new DefaultLiteral());
+        }
+        return qualifiers;
+    }
+
+    private static class DefaultLiteral extends AnnotationLiteral<Default> implements Default {
```

Comment:
Shouldn't be needed from CDI 2.0
It is now built in - https://github.com/eclipse-ee4j/cdi/blob/master/api/src/main/java/javax/enterprise/inject/Default.java#L104

Reply:
If I remove that I am getting ambiguous dependencies on injecting the application `MetricRegistry` from https://github.com/smallrye/smallrye-metrics/blob/2.1.4/implementation/src/main/java/io/smallrye/metrics/MetricRegistries.java#L40

Reply:
You should use `Default.Literal.INSTANCE` instead...

Reply:
Ok will use that one

Reply:
Ah, ok